crowbar_framework/app/models/node_object.rb

```diff
@@ -1346,9 +1346,9 @@ def unique_name_already_claimed_by(device)
   # that means that the given linkname can be found in the array
   # @node[:block_device][device][by-{id,path,uuid}]
   def link_to_device?(device, linkname)
-    # device is i.e. "/dev/sda", "/dev/vda", ...
+    # device is i.e. "sda", "vda", ...
     # linkname is i.e. "/dev/disk/by-path/pci-0000:00:04.0-virtio-pci-virtio1"
-    return true if device == linkname
+    return true if File.join("/dev/", device) == linkname
```

Comment:
> If you want to use File.join here, I think File.join("", "dev", device) would be more correct. I'd be fine with "/dev/#{device}" though.

Fixed in https://github.com/crowbar/barclamp-crowbar/commit/aedda153e73fd60557ce99e097c1ca307b6862b6